pkg/operator/operator.go

```diff
@@ -212,26 +214,30 @@ func New(config operatorconfig.Config, kubeConfig *rest.Config) (*Operator, erro
 		manager: mgr,
 		// TODO: These are only needed for the default ingress controller stuff, which
 		// should be refactored away.
-		client:    mgr.GetClient(),
-		namespace: config.Namespace,
+		client:                          mgr.GetClient(),
+		namespace:                       config.Namespace,
+		disableDefaultIngressController: config.DisableDefaultIngressController,
 	}, nil
 }
 
 // Start creates the default IngressController and then starts the operator
 // synchronously until a message is received on the stop channel.
 // TODO: Move the default IngressController logic elsewhere.
 func (o *Operator) Start(ctx context.Context) error {
-	// Periodicaly ensure the default controller exists.
-	go wait.Until(func() {
-		if !o.manager.GetCache().WaitForCacheSync(ctx) {
-			log.Error(nil, "failed to sync cache before ensuring default ingresscontroller")
-			return
-		}
-		err := o.ensureDefaultIngressController()
-		if err != nil {
-			log.Error(err, "failed to ensure default ingresscontroller")
-		}
-	}, 1*time.Minute, ctx.Done())
+
+	if !o.disableDefaultIngressController {
+		// Periodicaly ensure the default controller exists.
+		go wait.Until(func() {
+			if !o.manager.GetCache().WaitForCacheSync(ctx) {
+				log.Error(nil, "failed to sync cache before ensuring default ingresscontroller")
+				return
+			}
+			err := o.ensureDefaultIngressController()
+			if err != nil {
+				log.Error(err, "failed to ensure default ingresscontroller")
+			}
+		}, 1*time.Minute, ctx.Done())
+	}
```

Comment:
One more thing. It'd probably be helpful to have a INFO level log here to say it's skipping creation of the default ingress controller.
`log.Info("skipping default ingress controller creation")` or something

Reply:
updated